extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/AuthenticationProviderService.java

```diff
@@ -326,7 +354,7 @@ public LDAPAuthenticatedUser authenticateUser(Credentials credentials)
      * @throws GuacamoleException
      *     If an error occurs retrieving the user DN or the attributes.
      */
-    private Map<String, String> getAttributeTokens(ConnectedLDAPConfiguration config)
+    private Map<String, String> getAttributeTokens(ConnectedLDAPConfiguration config, Credentials credentials)
```

Comment:
This function is specifically documented as:

> Returns parameter tokens generated from LDAP attributes on the user currently bound under the given LDAP connection.

and is specifically named `getAttributeTokens()`. So long as that's the case, it's bad practice to let injection of an entirely different kind of token be a side-effect of this function.

I do see that the documentation here has been updated to clarify this a bit, however that ultimately resulted in the following statement:

> The attributes to be converted into parameter tokens must be explicitly listed in guacamole.properties or domain name of the LDAP connection when multiple auth enabled.

As-written, this does not make sense to me, and the original issues with primary documented scope and the naming of the function remain. If this function is to serve as the basis for injection of the `LDAP_DOMAINS` token (which I agree may make sense), then:

* The function should be renamed to capture its now-broadened scope (`getAttributeTokens()` is no longer correct).
* The function's documentation should be updated to capture that scope ("Returns parameter tokens generated from LDAP attributes" is no longer correct, and other documented aspects may also need adjusting).

Makes sense?

For example, if the function were renamed to something like `getUserTokens()`, the following might be reasonable:

> Returns parameter tokens generated based on details specific to the user currently bound under the given LDAP connection. Both LDAP attributes on the user's associated LDAP object and the credentials submitted by the user to Guacamole are considered. If any tokens are to be derived from LDAP attributes, those attributes must be explicitly listed in guacamole.properties. If no tokens are applicable, an empty map is returned.